src/include/txn_inline.h

```diff
@@ -98,6 +98,32 @@ __wt_txn_op_set_recno(WT_SESSION_IMPL *session, uint64_t recno)
     op->u.op_col.recno = recno;
 }
 
+/*
+ * __txn_op_need_set_key --
+ *     Check if we need to copy the key to the most recent transaction operation.
+ */
+static WT_INLINE bool
+__txn_op_need_set_key(WT_TXN *txn, WT_TXN_OP *op)
+{
+    /*
+     * We save the key for resolving the prepared updates. However, if we have already set the
+     * commit timestamp, the transaction cannot be prepared. Therefore, no need to save the key.
+     */
+    if (F_ISSET(txn, WT_TXN_HAS_TS_COMMIT))
+        return (false);
+
```

Comment:
Thanks for moving this into its own function. Can we add some clarifying comments?
"Prepared updates cannot exist in the history store."
"The metadata file cannot be prepared."
"Autocommit transactions cannot have prepare called on them."